### PR TITLE
682 pets component

### LINKED ISSUE #682: Other Pets component
### Dependency
- Awaiting updated design

### Overview
Implement the "Pets" component for the Intake profile.

Screenshot 2024-04-23 at 5 46 15 PM

You'll have to work off the `feat/generated-profile` branch so make sure to check it out and pull in the most up-to-date changes.

### Action Items

- [x] Add a "pets" type to the FieldTypes type definition.
- [x] Add a case to the switch statement that renders fields
- [x] Add the field to the `/src/utils/test/db/profile.ts` in the "Other Guests/Pets section"
- [x] Build the pets field component:
  - [x] Create a component that renders a field for a user to type in a pet
  - [x] The field has a button that allows the user to add another pet
  - [x] Add a button to delete pets
  - [x] Add custom validation if necessary

### Resources/Instructions
The `AdditionalGuestsField` could be a good reference to how to approach building this, but you will have to make some tweaks. You may have to define the data as an array of object so [here](https://formik.org/docs/guides/arrays) is some info from Formik on how to handle that.

### PR DESCRIPTION
**Pull request by johnwroge**
Closes #682 

### What changes did you make?

I added a 'pets' type to the FieldTypes type definition, updated the IntakeProfile directory to include a case of 'pets' to display the form elements, added the field to profile.ts file in the services directory, and built an additional pets field component 'AdditionalPetsField' which allows the user to add/remove pets. This component contains a search bar and is limiited to a group of pets and can updated to include other pets as necessary.  
  

### Rationale behind the changes?

 The intake profile section needs a pets component for guests so they can add any pets they own to their intake profile. This will help coordinators make appropriate matches with hosts who may/may not have preferences for pets. 
  
### Testing done for these changes
  
Tests were run in the app directory with all test cases passing. 
  
### What did you learn or can share that is new?(optional)
 
 I learned about representing form components in an alternative way. 
  

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>
325033528-041c485b-ad7d-4a08-be5b-29b76857cd85

image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
image
Screen Shot 2024-06-02 at 2 06 19 PM
Screen Shot 2024-06-02 at 2 07 02 PM

</details>

### CODE REVIEW

File: app/src/main.tsx [hunk 1]
```diff
@@ -148,9 +148,10 @@ function HuuApp() {
         <Route
           path="/profile/:profileId"
           element={
-            <ProtectedRoute>
-              <IntakeProfile />
-            </ProtectedRoute>
+            // <ProtectedRoute>
+            //   <IntakeProfile />
+            // </ProtectedRoute>
+            <IntakeProfile />
```
Comment (erikguntner) [inline app/src/main.tsx:154]: Please uncomment the protected route.

File: app/src/utils/test/db/profile.ts [hunk 2]
```diff
@@ -96,13 +96,22 @@ export const intakeProfiles: GetProfileApiResponse[] = [
           },
           {
             id: faker.string.numeric(4),
-            title: 'Pets',
+            title: 'Do you have any pets?',
```
Comment (erikguntner) [inline app/src/utils/test/db/profile.ts:99]: I believe it should be "Do you have any pets in your care?

**Review by erikguntner — APPROVED** [review]
Thanks @johnwroge this looks good! Just a few minor fixes and this should be good to go.

File: app/src/main.tsx (hunk 1 above)
Comment (johnwroge) [reply app/src/main.tsx]: Thanks Erik, I removed the commented protected route.

File: app/src/utils/test/db/profile.ts (hunk 2 above)
Comment (johnwroge) [reply app/src/utils/test/db/profile.ts]: Thanks for catching this. I updated it.

**Pull request merged**